packages/native/examples/with-distinct-filed-prop/App.js

```diff
@@ -0,0 +1,275 @@
+import React, { useState, useRef } from 'react';
+import {
+  SearchBase,
+  SearchComponent,
+  SearchBox
+} from '@appbaseio/react-native-searchbox';
+import { AntDesign } from '@expo/vector-icons';
+import {
+  StyleSheet,
+  Text,
+  View,
+  Platform,
+  StatusBar,
+  ActivityIndicator,
+  FlatList,
+  Image,
+  SafeAreaView,
+  ScrollView
+} from 'react-native';
+
+const renderResultItem = ({ item }) => {
+  const otherBooks = item.inner_hits
+    ? item.inner_hits.other_books.hits.hits
+    : [];
+  const updatedBooks = otherBooks.filter(
+    book => book._source.original_title !== item.original_title
+  );
+  // console.log('updatedBooks ==>> ', updatedBooks);
+  return (
+    <View style={styles.itemStyle}>
+      <Image
+        style={styles.image}
+        source={{
+          uri: item.image
+        }}
+        resizeMode="contain"
+      />
+      <ScrollView style={{ flex: 1 }}>
+        <Text style={styles.textStyle}>{item.original_title}</Text>
+        <Text style={styles.textStyle}>by {item.authors}</Text>
+        <View style={styles.star}>
+          {Array(item.average_rating_rounded)
+            .fill('x')
+            .map((i, index) => (
+              <AntDesign
+                key={item._id + `_${index}`}
+                name="star"
+                size={24}
+                color="gold"
+              />
+            ))}
+          <Text style={styles.rating}>({item.average_rating} avg)</Text>
+        </View>
+        <Text>Pub {item.original_publication_year}</Text>
+        {updatedBooks.length > 0 && (
+          <View>
+            <Text style={styles.otherBooksHeader}>
+              Other books by the same Author(s) :
+            </Text>
+            <View style={styles.row}>
+              {updatedBooks.map(book => (
+                <View key={book._id} style={styles.col}>
+                  <View style={styles.otherBooksContainer}>
+                    <Image
+                      style={styles.otherBooksImage}
+                      source={{
+                        uri: book._source.image
+                      }}
+                      resizeMode="contain"
+                    />
+                    <Text style={styles.otherBooksText}>
+                      {book._source.original_title.length > 12
```

Comment:
@anik-ghosh-au7 You can avoid this check. `substring` method is safe to be used

Reply:
used `numberOfLines` prop